RFS/src/main/java/com/rfs/ReindexFromSnapshot.java

```diff
@@ -0,0 +1,298 @@
+package com.rfs;
+
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.List;
+
+import com.beust.jcommander.JCommander;
+import com.beust.jcommander.Parameter;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+import org.apache.lucene.document.Document;
+import org.apache.logging.log4j.Level;
+import org.apache.logging.log4j.Logger;
+import org.apache.logging.log4j.LogManager;
+
+import com.rfs.common.*;
+import com.rfs.transformers.*;
+import com.rfs.version_es_6_8.*;
+import com.rfs.version_es_7_10.*;
+import com.rfs.version_os_2_11.*;
+
+public class ReindexFromSnapshot {
+    private static final Logger logger = LogManager.getLogger(ReindexFromSnapshot.class);
+
+    public static class Args {
+        @Parameter(names = {"-n", "--snapshot-name"}, description = "The name of the snapshot to read", required = true)
+        public String snapshotName;
+
+        @Parameter(names = {"--snapshot-dir"}, description = "The absolute path to the source snapshot directory on local disk", required = false)
+        public String snapshotDirPath = null;
+
+        @Parameter(names = {"--s3-local-dir"}, description = "The absolute path to the directory on local disk to write S3 files to", required = false)
+        public String s3LocalDirPath = null;
+
+        @Parameter(names = {"--s3-repo-uri"}, description = "The S3 URI of the snapshot repo, like: s3://my-bucket/dir1/dir2", required = false)
+        public String s3RepoUri = null;
+
+        @Parameter(names = {"--s3-region"}, description = "The AWS Region the S3 bucket is in, like: us-east-2", required = false)
+        public String s3Region = null;
+
+        @Parameter(names = {"-l", "--lucene-dir"}, description = "The absolute path to the directory where we'll put the Lucene docs", required = true)
+        public String luceneDirPath;
+
+        @Parameter(names = {"-h", "--target-host"}, description = "The target host and port (e.g. http://localhost:9200)", required = true)
+        public String targetHost;
+
+        @Parameter(names = {"-u", "--target-username"}, description = "The target username", required = true)
```

Comment:
We may want to make username and password not required to support clusters which have no auth